airbyte-cdk/python/airbyte_cdk/sources/declarative/async_job/job_orchestrator.py

```diff
@@ -111,8 +111,12 @@ def _start_jobs(self) -> None:
             self._replace_failed_jobs(partition)
```

Comment:
We are probably also starting jobs when we replace failed job so there is probably something to be done here too. 

I'm wondering if we should catch errors during the polling process as well... 🤔

Reply:
refactored

Reply:
Can we document why or why not we should have the try/catch on the update status as well?